### PR TITLE
migrate command should migrate configs sequentially

### PR DESCRIPTION
**Pull request by mlachkar**
…spath for Test

- In fact Test configuration relies on Compile to retrieve dependencyClasspath:
which implies that the src code needs to compile first in Scala 3 before migrating tests

### CODE REVIEW

File: plugin/src/main/scala/migrate/TypeInferenceMigration.scala [hunk 1]
```diff
@@ -1,43 +1,49 @@
...
   val internalImpl = Def.taskDyn {
-    val projectRef = thisProjectRef.value
-    val configs    = Keys.migrationConfigs.value
-    val sv         = scalaVersion.value
-
-    val projectId = projectRef.project
+    implicit val projectRef = thisProjectRef.value
+    val configs             = Keys.migrationConfigs.value
+    val sv                  = scalaVersion.value
+    implicit val projectId  = projectRef.project
```
Comment (adpi2) [inline plugin/src/main/scala/migrate/TypeInferenceMigration.scala:20]: This has nothing to do with the PR, but I really don't like that the formatter aligns everything. It adds a lot of changes that are not real changes.

File: plugin/src/sbt-test/sbt-scala3-migrate/type-inference-migration/build.sbt [hunk 2]
```diff
@@ -2,6 +2,8 @@ lazy val `type-inference-migration` = project
   .in(file("."))
   .settings(
     scalaVersion := "2.13.5",
+    libraryDependencies +=
+      "ch.epfl.scala" %% "scalafix-rules" % "0.9.26" % Test cross CrossVersion.for3Use2_13, // to create a test configuration
```
Comment (adpi2) [inline plugin/src/sbt-test/sbt-scala3-migrate/type-inference-migration/build.sbt:6]: I don't think this is needed. `Test` is a default config in sbt, it is always created.

**Review by adpi2 — APPROVED** [review]
LGTM!

File: plugin/src/main/scala/migrate/TypeInferenceMigration.scala (hunk 1 above)
Comment (mlachkar) [reply plugin/src/main/scala/migrate/TypeInferenceMigration.scala:20]: maybe the implicits ?

File: plugin/src/sbt-test/sbt-scala3-migrate/type-inference-migration/build.sbt (hunk 2 above)
Comment (mlachkar) [reply plugin/src/sbt-test/sbt-scala3-migrate/type-inference-migration/build.sbt:6]: ah ok!

**Pull request merged**